### ISSUE TITLE
Why GPU memory increases a lot in the silhouette renderer

### ISSUE THREAD

**Issue by pengsongyou** [facebookresearch/pytorch3d#569]
Hi,

Thanks again for the amazing library! I have been using your amazing silhouette renderer but recently encountered an issue.

Let's take your [fit_textured_mesh](https://github.com/facebookresearch/pytorch3d/blob/master/docs/tutorials/fit_textured_mesh.ipynb) tutorial as an example. You average the silhouette loss over 2 randomly sampled views as below:
```Python
for j in np.random.permutation(num_views).tolist()[:num_views_per_iteration]:
        images_predicted = renderer_silhouette(new_src_mesh, cameras=target_cameras[j], lights=lights)
        predicted_silhouette = images_predicted[..., 3]
        loss_silhouette = ((predicted_silhouette - target_silhouette[j]) ** 2).mean()
        loss["silhouette"] += loss_silhouette / num_views_per_iteration
```
However, since my `num_views_per_iteration` is larger than 2, I feel that it is quite inefficient to run the renderer to obtain silhouette for every single view. Therefore, what I did was, instead of have a list of camera in `target_cameras`, I combine all cameras in a batch: 
```Python
R = torch.cat([cam.R for cam in target_cameras], dim=0)
T = torch.cat([cam.T for cam in target_cameras], dim=0)
CAM = OpenGLPerspectiveCameras(R=R, T=T, device=device)
images_predicted = renderer_silhouette(mesh.extend(num_views_per_iteration), cameras=CAM, lights=lights)
```
Doing so indeed speeds up the process a lot (in my case it is >10-30 times faster). However, this leads to much higher GPU memory (in my case, if we consider 6 views, the GPU memory usage was increased 2G - 3G). I am wondering how we can handle this memory issue while keeping the memory low?

Best,
Songyou

**Comment by gkioxari** [comment]
It makes sense why the memory is increasing. You are allocating tensor of bigger size (due to the bigger batch size = num_views in your case). If you want to use less memory then you can use a smaller batch size.

You should read this note: https://github.com/facebookresearch/pytorch3d/blob/master/docs/notes/renderer.md

**Comment by pengsongyou** [comment]
Hi @gkioxari, thanks so much for your rapid reply. 

Yes, it makes sense that the memory increase when having a larger size. However, I still don't understand: if we keep batch size = 1 but loop over `num_views_per_iteration`, why the memory does not increase? The intermediate gradients on vertices from all views are kept as well, which seems like increasing the batch size, no? 

Thanks again!

**Comment by gkioxari** [comment]
No it's not the same. There is a lot of things under the PyTorch hood when computing forward/backward passes in a graph. Memory is allocated based on that graph, tensors for each variable need to be contiguous etc etc.

**Comment by pengsongyou** [comment]
@gkioxari Thanks so much for your answer! So we basically either go with faster inference but more memory, or slow inference but lower memory.

Thanks a lot again!
